src/batchhafnian.hpp

```diff
@@ -0,0 +1,174 @@
+// Copyright 2019 Xanadu Quantum Technologies Inc.
+
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+#include <stdafx.h>
+
+#include <Eigen/Eigenvalues>
+#include <Eigen/LU>
+
+typedef unsigned long long int ullint;
+
+
+ullint vec2index(std::vector<int> &pos, int resolution) {
+    int dim = pos.size();
+    ullint nextCoordinate = 0;
+
+    nextCoordinate = pos[0]-1;
+    for(int ii = 0; ii < dim-1; ii++) {
+        nextCoordinate = nextCoordinate*resolution + (pos[ii+1]-1);
+    }
+
+    return nextCoordinate;
+
+}
+
+ullint factorial(int n)
```

Comment:
What is the largest factorial that you can calculate in integer precision ```ullint```

Reply:
I am not too sure about that. But factorial of 21 exceeds the range of `unsigned long long int` so I suppose the max would be 20.